### PR TITLE
{chem}[foss/2023a] MBX v1.1.0

### PR DESCRIPTION
**Pull request by pavelToman**
(created using `eb --new-pr`)
resolves https://github.com/vscentrum/vsc-software-stack/issues/377

### CODE REVIEW

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by pavelToman** [comment]
> Test report by @boegelbot **FAILED** Build succeeded for 0 out of 1 (1 easyconfigs in total) cns2 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz (haswell), Python 3.6.8 See https://gist.github.com/boegelbot/17c7b6f4ab9ded76a99f0a0f2c01da5e for a full test report.

Low on resources.
On node4002.donphan.os - Linux RHEL 8.8 (Ootpa), x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz, 1 x NVIDIA NVIDIA A2, 545.23.08, Python 3.11.3 the `maxparallel = 3` is ok with 8 cores.

File: easybuild/easyconfigs/m/MBX/MBX-1.1.0-foss-2023a.eb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+checksums = ['0f55f4950226defb46fd0814ad97f906ce4ffd6403af6817bd98cb3c68996692']
+
+builddependencies = [('Autotools', '20220317')]
+dependencies = [
+    ('Python', '3.11.3'),
+    ('SciPy-bundle', '2023.07'),
+    ('GSL', '2.7'),
+]
+
+buildininstalldir = True
```
[the remark this answers is not in the document]
Comment (pavelToman) [reply easybuild/easyconfigs/m/MBX/MBX-1.1.0-foss-2023a.eb:23]: I can use `MakeCp`, could you please elucidate me what is wrong with the `buildininstalldir = True`?
I've never use `buildininstalldir` before, I got the easyconfig with this var from @boegel.

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

File: easybuild/easyconfigs/m/MBX/MBX-1.1.0-foss-2023a.eb (hunk 1 above)
[the remark this answers is not in the document]
Comment (jfgrimm) [reply easybuild/easyconfigs/m/MBX/MBX-1.1.0-foss-2023a.eb:23]: Using `buildininstalldir` will, as it suggests, unpack and run the configure/build/etc. steps in the install directory.
Depending on the kind of filesystem used to host the install dir, that can be rather slow, especially compared to building on local SSD or in memory.

For builds that don't do configuration that necessitates building in the install dir, it's usually best to avoid doing so

[the remark this answers is not in the document]
Comment (pavelToman) [reply easybuild/easyconfigs/m/MBX/MBX-1.1.0-foss-2023a.eb:23]: Is it ok to stay with easyblock `ConfigureMake`, not use `buildininstalldir = True` and in `postinstallcmds` copy whole MBX-1.1.0 directory to the `installdir`?

[the remark this answers is not in the document]
Comment (boegel) [reply easybuild/easyconfigs/m/MBX/MBX-1.1.0-foss-2023a.eb:23]: I'll admit that I got a bit lazy here, and @jfgrimm's remark about building on the filesystem where the installation will be located can be quite slow is very valid.
I'm not sure how relevant it is in this case though, because although the installation takes a long time, it's mostly because the C++ compiler is chewing on individual C++ source files for a very long time, so I/O is a non-issue as far as I could tell.

If we don't enable `buildininstalldir` here, we would need to copy the subdirectory that has the Python bindings after the installation, that could be done via `postinstallcmds` indeed.

I wouldn't copy the whole build directory, only the extra bits that are necessary.

Part of the reason for this approach is that to build the LAMMPS plugin for MBX, you need to copy over some source files from MBX into LAMMPS, see https://github.com/paesanilab/MBX#lammps.

So, in short, I think we can:
* not enable `buildininstalldir`
* copy both `plugins/lammps/USER-MBX` via `postinstallcmds` to the installation directory, as follows:
```python
postinstallcmds = [
    "mkdir -p %(installdir)s/plugins/{lammps,python}",
    "cp -a plugins/lammps/USER-MBX %(installdir)s/plugins/lammps/",
    "cp -a plugins/python/mbx %(installdir)s/plugins/python/",
]
* add `plugins/lammps/USER-MBX` to `sanity_check_paths`

File: easybuild/easyconfigs/m/MBX/MBX-1.1.0-foss-2023a.eb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+preconfigopts = 'echo HEEERE && echo %(installdir)s && export MBX_HOME=%(installdir)s && autoreconf -fi && '
```
Comment (boegel) [inline easybuild/easyconfigs/m/MBX/MBX-1.1.0-foss-2023a.eb:25]: @pavelToman There's some debug stuff in here to clean up?

File: easybuild/easyconfigs/m/MBX/MBX-1.1.0-foss-2023a.eb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+configopts = '--enable-shared --enable-verbose CXX="$CXX"'
+
+prebuildopts = 'export MBX_HOME=%(installdir)s && '
+
+preinstallopts = 'export MBX_HOME=%(installdir)s && '
```
Comment (boegel) [inline easybuild/easyconfigs/m/MBX/MBX-1.1.0-foss-2023a.eb:31]: I'm not sure that setting `MBX_HOME` during the installation is needed at all, if not please remove

In any case, if `buildininstalldir` is not used, `MBX_HOME` should not be set to `installdir` during configure/build...

File: easybuild/easyconfigs/m/MBX/MBX-1.1.0-foss-2023a.eb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+runtest = 'check'
+
+modextrapaths = {'PYTHONPATH': '%(name)s-%(version)s/plugins/python'}
+
+modextravars = {'MBX_HOME': '%(installdir)s'}
+
+sanity_check_paths = {
+    'files': ['bin/mb_decomp', 'bin/optimize', 'bin/order_frames', 'bin/single_point',
+              'lib/libmbx.a', 'lib/libmbx.%s' % SHLIB_EXT],
+    'dirs': ['include', '%(name)s-%(version)s/plugins/python/mbx'],
```
Comment (boegel) [inline easybuild/easyconfigs/m/MBX/MBX-1.1.0-foss-2023a.eb:44]: The `%(name)s-%(version)s` bit is annoying here, that won't be needed if we don't use `buildininstalldir`

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3119.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/boegel/eba8b558b439d2d6a99f3d17431eda41 for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Going in, thanks @pavelToman!
